src/tests/unit/tests/bug-filing/azure-boards/azure-boards-bug-filing-service.test.tsx

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { shallow } from 'enzyme';
+import * as React from 'react';
+import { IMock, Mock } from 'typemoq';
+
+import {
+    AzureBoardsBugFilingService,
+    AzureBoardsBugFilingSettings,
+    AzureBoardsIssueDetailField,
+} from '../../../../../bug-filing/azure-boards/azure-boards-bug-filing-service';
+import { AzureBoardsSettingsForm } from '../../../../../bug-filing/azure-boards/azure-boards-settings-form';
+import { SettingsFormProps } from '../../../../../bug-filing/types/settings-form-props';
+import { UserConfigMessageCreator } from '../../../../../common/message-creators/user-config-message-creator';
+import { BugServicePropertiesMap } from '../../../../../common/types/store-data/user-configuration-store';
+
+describe('AzureBoardsBugFilingServiceTest', () => {
+    let userConfigMessageCreatorMock: IMock<UserConfigMessageCreator>;
+    let props: SettingsFormProps<AzureBoardsBugFilingSettings>;
+    let projectStub: string;
+    let issueDetailsLocationStub: AzureBoardsIssueDetailField;
+
+    const invalidTestSettings: AzureBoardsBugFilingSettings[] = [
+        null,
+        {} as AzureBoardsBugFilingSettings,
+        undefined,
+        { projectURL: '' } as AzureBoardsBugFilingSettings,
+        { projectURL: '', issueDetailsField: '' as AzureBoardsIssueDetailField },
+        { projectURL: 'some project', issueDetailsField: '' as AzureBoardsIssueDetailField },
+        { projectURL: '', issueDetailsField: 'some issue details location' as AzureBoardsIssueDetailField },
+    ];
+
+    beforeEach(() => {
+        projectStub = 'some project';
+        issueDetailsLocationStub = 'some location' as AzureBoardsIssueDetailField;
+        userConfigMessageCreatorMock = Mock.ofType(UserConfigMessageCreator);
+        props = {
+            deps: {
+                userConfigMessageCreator: userConfigMessageCreatorMock.object,
+            },
+            settings: {
+                projectURL: 'some project',
```

Comment:
Should we use the stub above to prevent divergence here? (Similarly for `issuesDetailsField` on next line?)
```suggestion
                projectURL: projectStub,
```

Reply:
Updated